src/middleware.js

```diff
@@ -8,10 +8,10 @@ export function middleware(request) {
   //   console.log(response.cookies, appSid, isAuthenticated, isUser, 222);
 
   if (!isAuthenticated) {
-    return NextResponse.rewrite(new URL('/login/', request.url));
+    return NextResponse.redirect(new URL('/login/', request.url));
```

Comment:
The apps should serve the login component, not redirect. This is done so the user returns to where they were editing on the spot.

Reply:
the `/login` contains the login component @glespinosa

Reply:
For example @darwin808 and @glespinosa if a user is on /teams and they are logged out, /teams should load the login component NOT redirect them to /login

Reply:
Same should apply for /instance /profile etc

Reply:
@ardeay 
### Trying to render a JSX Component(Login) in the middleware is causing error maybe because the middleware is not accepting react components, looks like Redirecting them is the only option for now.
Error: ![image](https://user-images.githubusercontent.com/44116036/188834299-3e490f09-6e1a-4aaa-a7aa-f2bd82faf4b8.png)

In the refactored middleware code below the behavior are as follows:

If logged in:
 - If user tried to access unprotected routes it will automatically redirect to /instances/

If logged out:
 - If user tried to access protected routes, he will be redirected to the /login